pandas/core/missing.py

```diff
@@ -452,138 +452,67 @@ def interpolate_2d(values, method='pad', axis=0, limit=None, fill_value=None,
     return values
 
 
-def _interp_wrapper(f, wrap_dtype, na_override=None):
-    def wrapper(arr, mask, limit=None):
-        view = arr.view(wrap_dtype)
-        f(view, mask, limit=limit)
-
-    return wrapper
-
-
-_pad_1d_datetime = _interp_wrapper(algos.pad_inplace_int64, np.int64)
-_pad_2d_datetime = _interp_wrapper(algos.pad_2d_inplace_int64, np.int64)
-_backfill_1d_datetime = _interp_wrapper(algos.backfill_inplace_int64, np.int64)
-_backfill_2d_datetime = _interp_wrapper(algos.backfill_2d_inplace_int64,
-                                        np.int64)
+def _cast_values_for_fillna(values, dtype):
+    """
+    Cast values to a dtype that algos.pad and algos.backfill can handle.
+    """
+    # TODO: for int-dtypes we make a copy, but for everything else this
```

Comment:
no idea why this would be. Would really try to be consistent. I think copy is more sematically correct for these types of routines, as we can't always guaranteee inplace behavior (meaning we *may* have to copy sometimes). So this may have some uncessary copying going on, but willing to trade that for better semantics.

Reply:
absolutely. I am ok with inplace too. Its just then the guarantee to the caller will have to be stronger. We probably don't have enough tests in place to make a strong statement either way, but go for it. This looks like a nice re-factor, so could address the inplace/copy question later (as long as we are pretty sure its not actualy mutating things in the caller, unless its specifically allowed).